lua/rustaceanvim/config/init.lua

```diff
@@ -110,6 +110,7 @@ vim.g.rustaceanvim = vim.g.rustaceanvim
 ---@class RustaceanLspClientOpts
 ---@field auto_attach? boolean | fun(bufnr: integer):boolean Whether to automatically attach the LSP client. Defaults to `true` if the `rust-analyzer` executable is found.
 ---@field cmd? string[] | fun():string[] Command and arguments for starting rust-analyzer
+---@field root_dir? fun(filename: string):string|nil Should return the directory to use for the attached LSP, or nil if no server should attach. Defaults to `rustaceanvim.cargo.get_root_dir`.
```

Comment:
Hmm... I think it would be good to somehow export the default version of this function (maybe exported from a different place with a different name, like `default_root_dir`), so that people can override it in a few cases, but otherwise fall back to the default behavior.

Reply:
I'd say yagni for now.
But an option could be for the function to take a second parameter: `default: fun(filename: string):string|nil`